MetaMask/metamask-extension
# fix: Fix getProviderState networkVersion for request queue

## **Description**

Currently the globally selected network's networkId is being sent as the `networkVersion` to all dapps when request queue is enabled. This PR fixes this by making an appropriate `net_version` call when each dapp's inpage provider requires it.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23590?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2176

## **Manual testing steps**

1. Enable request queue
2. Go to a dapp (Refresh the page if you had it opened before request queue was on)
3. use `wallet_switchEthereumChain` to make the dapp and globally selected networks different
3. check `window.ethereum.chainId` and `window.ethereum.networkVersion` values, they should match the dapp's network
1. Disable request queue
2. Go to a dapp (Refresh the page if you had it opened before request queue was on)
3. check `window.ethereum.chainId` and `window.ethereum.networkVersion` values, they should match the global network

## **Screenshots/Recordings**



### **Before**

Request Queue On

https://github.com/MetaMask/metamask-extension/assets/918701/6edde8d0-c911-4349-91f2-e7b5a18b32a8


### **After**

Request Queue On

https://github.com/MetaMask/metamask-extension/assets/918701/3490e442-4562-458d-b138-59cb7c6f34b9

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
